main.tf

```diff
@@ -47,6 +47,56 @@ resource "azurerm_linux_function_app" "linux_function" {
     }
   }
 
+  # Check app_service_plan; for example, azurerm_app_service_plan.example.id
+
+  resource "azurerm_private_endpoint "linux_private_endpoint" {
+    count           = var.asp_os_type == "Linux" && var.app_service_plan_type == "PremiumV2" ? 1 : 0
+    name = var.private_endpoint
+    location = var.location
+    resource_group_name         = var.resource_group_name
+    subnet_id = azurerm_subnet.main.id
+
+    private_service_connection {
+      name = var.private_service_connection
+      private_connection_resource_id = azurerm_linux_function_app.linux_function.id
+
+      subresource_name = ["site"]
+      is_manual_connection = false
+    }    
+  }
+
+  resource "azurerm_private_endpoint "windows_private_endpoint" {
+    count           = var.asp_os_type == "Windows" && var.app_service_plan_type == "PremiumV2" ? 1 : 0
+    name = var.private_endpoint
+    location = var.location
+    resource_group_name         = var.resource_group_name
+    subnet_id = azurerm_subnet.main.id
+
+    private_service_connection {
+      name = var.private_service_connection
+      private_connection_resource_id = azurerm_windows_function_app.windows_function.id
+
+      subresource_name = ["site"]
+      is_manual_connection = false
+    }    
+  }
+  
+# Integrate with VNet
+resource "azurerm_app_service_virtual_network_swift_connection" "private_function_vnet_link" {
+  app_service_id = var.function_app_name.id
```

Comment:
Should this be conditional as well ?

Reply:
Yes, will address as per your suggestion.  Thanks!